### DISCUSSION TITLE
Upstream and bootstrap DNS question

### DISCUSSION THREAD

**Discussion by carlylemiii** [AdguardTeam/AdGuardHome#2920] [Q&A]
Is this a valid config? It is not clear to me if I should set all these bootstrap servers or if they are unnecessary or too many? What’s “proper” practice? Should I have all the IPs that correspond to my upstreams?

![265DA69D-460D-4C0B-8D11-B5F899D815A4](https://user-images.githubusercontent.com/29067983/113461526-9c5a7d00-93e2-11eb-83bb-0fdf7a3d28e3.png)

**Comment by ainar-g** [comment]
Hi!  What are you trying to achieve exactly?  Bootstrap addresses are basically only used to resolve the hosts in the upstreams section.

**Reply by carlylemiii** [reply]
> What are you trying to achieve exactly?

That's actually a great question. 😅 By default, AdGuard Home has `https://dns10.quad9.net/dns-query` as its upstream and `9.9.9.10`, `149.112.112.10`, `2620:fe::10`, and `2620:fe::fe:10` as its bootstrap addresses. Since I replaced that upstream with `https://dns.quad9.net/dns-query`, I figured it only made sense to change the bootstraps that corresponded with that, as well (from https://kb.adguard.com/en/general/dns-providers#standard-2).

But later I also added `https://dns.cloudflare.com/dns-query` as an upstream, so since the default originally had the corresponding addresses in the bootstrap section, I _thought_ maybe that I'd want the Cloudflare entries listed there as well.

Basically the meaning of the bootstraps is not completely clear to me, and I wasn't sure what I should have in that box after changing my upstream servers. Nothing? Those for only one of the upstreams? Everything, as in the screenshot?

Thanks for the help with a dumb question. 🙂

**Reply by ainar-g** [reply]
Most of the time, you don't really need to change the bootstrap addresses.  Again, they are only used to resolve the hostnames of the upstream servers (the ones you put into an input above).  Bootstrap servers must simply be IP addresses of available DNS servers.
